scripts/generate_example_pngs.R

```diff
@@ -0,0 +1,191 @@
+
+# Have a list of topics to be excluded from the image-generation
+# process; all of these don't produce `gt_tbl` objects
+name_exclusions <-
+  c(
+    "as.data.frame.gt_tbl",
+    "as_latex",
+    "as_raw_html",
+    "as_rtf",
+    "cells_data",
+    "cells_group",
+    "countrypops",
+    "escape_latex",
+    "gt-options",
+    "gt-package",
+    "gt_latex_dependencies",
+    "gt_output",
+    "gtcars",
+    "gtsave",
+    "pipe",
+    "pizzaplace",
+    "print.gt_tbl",
+    "random_id",
+    "reexports",
+    "render_gt",
+    "sp500",
+    "sza",
+    "test_image"
+  )
+
+# Revise the list of topic names by removing the excluded ones
+fn_names <- fn_names %>% base::setdiff(name_exclusions)
+
+# Create a `save_with_webshot()` function that allows us to
+# resize the images before writing to disk
+save_with_webshot <- function(data,
+                              filename,
+                              path = NULL,
+                              ...,
+                              zoom = 2,
+                              expand = 5,
+                              resize = NULL) {
+
+  gtsave_filename <- function(path, filename) {
+
+    if (!is.null(path)) {
+      filename <- file.path(path, filename)
+    }
+
+    filename %>% gt:::path_expand()
+  }
+
+  filename <- gtsave_filename(path = path, filename = filename)
+
+  # Create a temporary file with the `html` extension
+  tempfile_ <- tempfile(fileext = ".html")
+
+  # Reverse slashes on Windows filesystems
+  tempfile_ <-
+    tempfile_ %>%
+    gt:::tidy_gsub("\\\\", "/")
+
+  # Save gt table as HTML using the `gt_save_html()` function
+  data %>% gt:::gt_save_html(filename = tempfile_, path = NULL)
+
+  # Saving an image requires the webshot package; if it's
+  # not present, stop with a message
+  if (!requireNamespace("webshot", quietly = TRUE)) {
+
+    stop("The `webshot` package is required for saving images of gt tables.",
+         call. = FALSE)
+
+  } else {
+
+    if (is.null(resize)) {
+      webshot::webshot(
+        url = paste0("file:///", tempfile_),
+        file = filename,
+        zoom = zoom,
+        expand = expand,
+        ...
+      )
+    } else {
+      webshot::webshot(
+        url = paste0("file:///", tempfile_),
+        file = filename,
+        zoom = zoom,
+        expand = expand,
+        ...
+      ) %>%
+        webshot::resize(resize)
+    }
+  }
+}
+
+# Declare a fixed image width
+image_width <- 1100
+
+# Obtain a temporary directory name
+temp_dir <- tempdir()
+
+# Generate images in `temp_dir`
+for (fn_name in fn_names) {
+
+  # Get the initial list of objects in the environment
+  obj_list_initial <- ls()
+
+  # Evaluate the examples in the examples for each topic/function; the
+  # objects we'd want to create images from all begin with `tab_`
+  utils::example(topic = fn_name, package = "gt", character.only = TRUE, give.lines = FALSE)
+
+  # Get a vector of new objects created
+  new_obj <- base::setdiff(ls(), obj_list_initial)
```

Comment:
Use `local({ ... })` instead, to give yourself a clean sheet each time (rather than adding/removing stuff to the same environment for every iteration)

Reply:
Or lapply

Reply:
Not possible it seems but the `for` loop approach is better now at least.